src/errors/ApiKeyError.ts

```diff
@@ -0,0 +1,3 @@
+import { ValidationError } from './ValidationError';
+
+export class ApiKeyError extends ValidationError {}
```

Comment:
I don't think we need this to extend ValidationError (those are errors related to data structure validation)

I would probably just call this an `AuthenticationError` (and have it extend the built in `Error` object).